core/include/cubos/core/memory/stream.hpp

```diff
@@ -58,12 +58,26 @@ namespace cubos::core::memory
         /// @return Number of bytes read.
         virtual std::size_t read(void* data, std::size_t size) = 0;
 
+        /// @brief Reads data from the stream with the exact specified size.
+        /// @param data Buffer to read data into.
+        /// @param size Size of the buffer.
+        /// @note This function will only return after buffer is completed.
+        /// @return Whether reading was sucessful.
+        bool readExact(void* data, std::size_t size);
```

Comment:
I'm having a tough time understanding differences between this method and the 

`         virtual std::size_t read(void* data, std::size_t size) = 0;`

Besides what it returns what changes?

Reply:
It was because of this;:

![image](https://github.com/user-attachments/assets/457b9ebe-811d-4f93-927d-7734d4075a21)

but I am removing it, since Ive talked with @RiscadoA and other Stream implementations are assuming that `read` handles the whole buffer.